content/concepts/consensus-network/transaction-malleability.md

```diff
@@ -0,0 +1,142 @@
+
+2. The system notes the identifying hash of the vulnerable transaction, submits it to the XRP Ledger network, then begins monitoring for that hash to be included in a validated ledger version.
+
+3. A malicious actor sees the transaction propagating through the network before it becomes confirmed.
+
+4. The malicious actor calculates the alternate signature for the vulnerable transaction, and derives the hash of the resulting transaction data.
+
+    Unlike creating a signature for a different transaction data, this does not require a large amount of computational work. It can be done in much less time than it takes to generate a signature in the first place.
+
+5. The malicious actor submits the modified (likely non-fully-canonical) transaction to the network.
+
+    This creates a "race" between the transaction as originally submitted and the modified version submitted by the malicious actor. The two transactions are mutually exclusive. Both are valid, but they have the same exact transaction data, including the `Sequence` number, so at most one of them can ever be included in a validated ledger.
+
+    Servers in the peer-to-peer network have no way of knowing which one "came first" or was intended by its original sender. Delays or other coincidences in network connectivity could result in validators seeing only one or the other by the time they finalize their consensus proposals, so either one could "win the race".
+
+    A malicious actor could increase the chances of getting non-canonical transactions confirmed if they controlled some number of well-connected servers in the peer-to-peer network, even if those servers are not trusted as validators.
+
+    If the malicious actor controls the only server to which the vulnerable system submitted the transaction, the malicious actor can easily control which version is distributed to the rest of the network.
+
+6. The malicious actor's version of the transaction achieves a consensus and becomes included in a validated ledger.
+
+    At this point, the transaction has executed and cannot be reversed. Its effects (such as sending XRP) are final. The original version of the transaction is no longer valid because its `Sequence` number has been used.
+
+    The effects of the transaction in the XRP Ledger are exactly the same as if the original version had executed.
+
+7. The vulnerable system does not see the transaction hash it is expecting, and erroneously concludes that the transaction did not execute.
+
+    If the transaction included the `LastLedgerSequence` field, this would occur after the specified ledger index has passed.
+
+    If the transaction omitted the `LastLedgerSequence` field, this could be wrong in another way: there is no amount of time after which you can definitively conclude that such a transaction could not later succeed. (See [Reliable Transaction Submission](reliable-transaction-submission.html) for details.)
```

Comment:
I think this statement is not quite true?  If the user can confirm that the Sequence of the transaction has been consumed by the account (by, e.g., looking at the `Sequence` in `account_info`) then the user can conclude the original transaction can never succeed.  Or did I miss something?

Reply:
You're not really wrong, but the statement is not wrong either; it's supposed to mean, "Well, even if the transaction hadn't been included in a ledger at all, you maybe don't know it won't pull a [twenty-seventh amendment](https://en.wikipedia.org/wiki/Twenty-seventh_Amendment_to_the_United_States_Constitution) and gain a majority much, much later.

Reply:
Rough suggestion:
> If the transaction omitted the `LastLedgerSequence` field, this could be wrong in another way: in the absence of a different validated transaction consuming the same sequence number, there is no amount of time after which you can definitively conclude that such a transaction could not later succeed.

Reply:
I went with this:

> If the transaction omitted the `LastLedgerSequence` field, this could be wrong in another way: if no other transaction from the same sender uses the same `Sequence` number, then the transaction could theoretically succeed later regardless of how much time has passed.